File: src/models/units.rs
use serde::{Serialize, Deserialize};
use sqlx::types::{Decimal};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct UnitType {
    pub id: Uuid,
    pub ref_id: String,
    pub alt_id: Option<String>,
    pub name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Unit {
    pub id: Uuid,
    pub ref_id: String,
    pub alt_id: Option<String>,
    pub unit_type_id: Uuid,
    pub name: String,
    pub symbol: Option<String>,
    pub precision: Option<i32>,
    pub decimal_separator: String,
    pub thousands_separator: String,
    pub active: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct ConversionRate {
    pub id: Uuid,
    pub from_unit_id: Uuid,
    pub to_unit_id: Uuid,
    pub rate: Decimal,
}

File: internal/models/units.go
package models

import (
	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

type UnitType struct {
	ID         uuid.UUID  `json:"id" db:"id"`
	RefID      string     `json:"ref_id" db:"ref_id"`
	AltID      *string    `json:"alt_id" db:"alt_id"`
	Name       string     `json:"name" db:"name"`
	BaseUnitID *uuid.UUID `json:"base_unit_id" db:"base_unit_id"`
}

type Unit struct {
	ID                 uuid.UUID `json:"id" db:"id"`
	RefID              string    `json:"ref_id" db:"ref_id"`
	AltID              *string   `json:"alt_id" db:"alt_id"`
	UnitTypeID         uuid.UUID `json:"unit_type_id" db:"unit_type_id"`
	Name               string    `json:"name" db:"name"`
	Symbol             *string   `json:"symbol" db:"symbol"`
	Precision          *int32    `json:"precision" db:"precision"`
	DecimalSeparator   string    `json:"decimal_separator" db:"decimal_separator"`
	ThousandsSeparator string    `json:"thousands_separator" db:"thousands_separator"`
	Active             *bool     `json:"active" db:"active"`
}

type ConversionRate struct {
	ID         uuid.UUID       `json:"id" db:"id"`
	FromUnitID uuid.UUID       `json:"from_unit_id" db:"from_unit_id"`
	ToUnitID   uuid.UUID       `json:"to_unit_id" db:"to_unit_id"`
	Rate       decimal.Decimal `json:"rate" db:"rate"`
}
